compiler/modules/front.syntax/scan.bal

```diff
@@ -189,6 +189,13 @@ function unicodeEscapeValue(string fragment) returns string|error {
     return ch;
 }
 
+function scanLines(string[] lines) returns readonly & ScannedLine[] {
+    //JBUG NPE from BIROptimizer.java:159 when inlined
+    ScannedLine[] result = from var l in lines select scanLine(l);
+    // xxx don't clone, still haven't figured out how to use query to create readonly lists
+    return result.cloneReadOnly();
```

Comment:
@jclark what would be a good way to do this ? Is cloning ok?

Reply:
cloneReadOnly is fine.

The way it ought to work is that a query expression creates a read-only list when the contextually expected type is readonly.

Reply:
@gimantha please note.